src/tasks/component-installers/cert-manager.ts

```diff
@@ -0,0 +1,184 @@
+/*********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { Command } from '@oclif/command'
+import * as fs from 'fs'
+import * as Listr from 'listr'
+import * as os from 'os'
+import * as path from 'path'
+
+import { KubeHelper } from '../../api/kube'
+import { CERT_MANAGER_NAMESPACE_NAME, CHE_TLS_SECRET_NAME } from '../../constants'
+
+export const CERT_MANAGER_CA_SECRET_NAME = 'ca'
+
+export class CertManagerTasks {
+  protected kubeHelper: KubeHelper
+
+  constructor(flags: any) {
+    this.kubeHelper = new KubeHelper(flags)
+  }
+
+  /**
+   * Returns list of tasks which perform cert-manager checks and deploy and requests self-signed certificate for Che.
+   */
+  getTasks(flags: any, command: Command): ReadonlyArray<Listr.ListrTask> {
+    return [
+      {
+        title: 'Check Cert Manager deployment',
+        task: async (ctx: any, task: any) => {
+          // Check only one CRD of cert-manager assuming that it is installed or not.
+          ctx.certManagerInstalled = await this.kubeHelper.namespaceExist(CERT_MANAGER_NAMESPACE_NAME) && await this.kubeHelper.crdExist('certificates.cert-manager.io')
+          if (ctx.certManagerInstalled) {
+            task.title = `${task.title}...already deployed`
+          } else {
+            task.title = `${task.title}...not deployed`
+
+            return new Listr([
```

Comment:
It is not good to return tasks inside task.
It is better to skip unnecessary using `enable` property. And the flag can be put into ctx

Reply:
@tolusha I think it is a standard way to go. We may organize the sequence of tasks in more clear way: create a separate method which returns subtasks list. And that it will be clear to read (more we'll have displaying of the tasks as subtasks).
But if you really want to have flattened list of tasks, let it be so...